runtime/src/main/jni/ObjectManager.cpp

```diff
@@ -328,7 +328,17 @@ void ObjectManager::ReleaseJSInstance(Persistent<Object>* po, JSInstanceInfo* js
 
     m_idToObject.erase(it);
     m_released.insert(po, javaObjectID);
+
+    // In case the MarkReachableObjects is perfect, the regular instances will be reachable from an implementation object and won't be released.
+    // However, getting by mistake here, leads to referencing dangling pointers downstream, trying to get objects by weird ids.
+    // That's why we are clearing the external JSInstanceInfo* pointer from the JavaScript instance here.
```

Comment:
if somewhere down the road we try to get the information from `JsInfo` to help us debug the problem and improve the logic, that information will be gone, thus leaving us no information. I think we should leave the current logic to preserve information that can help us improve.